### ISSUE TITLE
Double-charging privacy in DC-GAN example

### ISSUE THREAD

**Issue by ffuuugor** [pytorch/opacus#418]
tl;dr: [dcgan.py](https://github.com/pytorch/opacus/blob/main/examples/dcgan.py) example adds noise to the fake data gradients, which it doesn't need to.
We should change the training pipeline (because it's more correct) and measure the impact (because we're curious).

See more details and solution proposals on the PyTorch forum [thread](https://discuss.pytorch.org/t/double-charging-privacy-in-dcgan-example/149098)

**Comment by Koukyosyumei** [comment]
@ffuuugor 

Hi! I'm Hideaki Takahashi, and currently working on this issue. I've made a PR but need approval from the maintainer to run the workflow. How should I proceed?

**Comment by romovpa** [comment]
@Koukyosyumei Thank you for contributing! 
We will review it soon. Let's discuss in your PR, I've left a comment.
